File: glip-team-messaging/chats/get-chats/code-samples/listGlipChats.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //type: [ 'Everyone', 'Group', 'Personal', 'Direct', 'Team' ],
    //recordCount: 30,
    //pageToken: '<ENTER VALUE>'
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.get(`/restapi/v1.0/glip/chats`, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});

File: glip-team-messaging/chats/get-chats/code-samples/listGlipChats.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
ListGlipChatsParameters listGlipChatsParameters = new ListGlipChatsParameters {
    //type = new[] { "Everyone", "Group", "Personal", "Direct", "Team" },
    //recordCount = 30,
    //pageToken = "<ENTER VALUE>"
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Glip().Chats().List(listGlipChatsParameters);
// PROCESS RESPONSE
